app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt

```diff
@@ -165,25 +165,31 @@ constructor(
     _suggestions.value = emptyMap()
   }
 
-  fun fetchActivities(rad: Double = 10000.0, centerLocation: LatLng = LatLng(46.519962, 6.633597)) {
+  fun fetchActivities() {
     viewModelScope.launch {
       _isLoading.value = true
-      _activities.value = ArrayList()
-      _activityIds.value = ArrayList()
+      val activitiesHolder: ArrayList<Activity> = ArrayList()
```

Comment:
You don't need to use temp lists here, you can just update the state value directly.

Reply:
In fact, I need to do it since in iterate through all the activity types in the for loop. Thus, I need to have a holder list that adds everything at each iteration, if I do what you are suggesting, I might have only the last iteration of the loop and not all the previous ones.